packages/relay/src/lib/validators/objectTypes.ts

```diff
@@ -192,153 +201,69 @@ export const OBJECTS_VALIDATIONS: { [key: string]: IObjectSchema } = {
   },
 };
 
-export class DefaultValidation<T extends object = any> implements IObjectValidation<T> {
-  private readonly _object: T;
-  protected readonly schema: IObjectSchema;
-
-  constructor(schema: IObjectSchema, object: T) {
-    this.schema = schema;
-    this._object = object;
-  }
-
-  get object(): T {
-    return this._object;
-  }
-
-  validate() {
-    if (this.schema.failOnUnexpectedParams) {
-      this.checkForUnexpectedParams();
-    }
-    if (this.schema.deleteUnknownProperties) {
-      this.deleteUnknownProperties();
-    }
-    return Validator.validateObject(this.object, this.schema);
-  }
-
-  name() {
-    return this.schema.name;
-  }
-
-  checkForUnexpectedParams() {
-    const expectedParams = Object.keys(this.schema.properties);
-    const actualParams = Object.keys(this.object);
+export function validateSchema(schema: IObjectSchema, object: any) {
+  if (schema.failOnUnexpectedParams) {
+    const expectedParams = Object.keys(schema.properties);
+    const actualParams = Object.keys(object);
     const unknownParam = actualParams.find((param) => !expectedParams.includes(param));
     if (unknownParam) {
-      throw predefined.INVALID_PARAMETER(`'${unknownParam}' for ${this.schema.name}`, `Unknown parameter`);
+      throw predefined.INVALID_PARAMETER(`'${unknownParam}' for ${schema.name}`, `Unknown parameter`);
     }
   }
-
-  deleteUnknownProperties() {
-    const expectedParams = Object.keys(this.schema.properties);
-    const actualParams = Object.keys(this.object);
+  if (schema.deleteUnknownProperties) {
+    const expectedParams = Object.keys(schema.properties);
+    const actualParams = Object.keys(object);
     const unknownParams = actualParams.filter((param) => !expectedParams.includes(param));
```

Comment:
nit: we could move these constants to the top of the function since they're reused for both if's